lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants.
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and sample
+  targets `y`.
+
+  Depending on number of classes the function chooses either binary
+  or multinomial logistic regression.
+  """
+  defn fit(x, y, opts \\ []) do
+    opts =
+      keyword!(opts,
+        iterations: 1000,
+        lr: 0.01,
+        num_classes: 1
+      )
+
+    fit_verify(x, y, opts)
+    if opts[:num_classes] < 3, do: fit_binary(x, y, opts), else: fit_multinomial(x, y, opts)
+  end
+
+  # Function checks validity of the provided data
+
+  deftransformp fit_verify(x, y, opts) do
+    unless is_integer(opts[:num_classes]) and opts[:num_classes] > 0 do
+      raise ArgumentError,
+            "expected :num_classes to be a positive integer, got: #{inspect(opts[:num_classes])}"
+    end
+
+    if Nx.rank(x) != 2 do
+      raise ArgumentError,
+            "expected x to have shape {n_samples, n_features}, got tensor with shape: #{inspect(Nx.shape(x))}"
+    end
+
+    if Nx.rank(y) != 1 do
+      raise ArgumentError,
+            "expected y to have shape {n_samples}, got tensor with shape: #{inspect(Nx.shape(y))}"
+    end
+
+    unless is_number(opts[:lr]) and opts[:lr] > 0 do
+      raise ArgumentError, "expected :lr to be a positive number, got: #{inspect(opts[:lr])}"
+    end
+
+    unless is_integer(opts[:iterations]) and opts[:iterations] > 0 do
+      raise ArgumentError,
+            "expected :iterations to be a positive integer, got: #{inspect(opts[:iterations])}"
+    end
+  end
+
+  # Binary logistic regression
+
+  defnp fit_binary(x, y, opts \\ []) do
+    iterations = opts[:iterations]
+    lr = opts[:lr]
+    x_t = Nx.transpose(x)
+    y_t = Nx.transpose(y)
+
+    {_m, n} = x.shape
```

Comment:
```suggestion
    {_m, n} = Nx.shape(x)
```

better to rely on the public function :)